client/src/controller/commandManager/DeleteSelectionCommand.hx

```diff
@@ -0,0 +1,61 @@
+package controller.commandManager;
+import model.component.CircuitDiagramI;
+import model.selectionModel.SelectionModel;
+import model.component.Component;
+import model.component.Link;
+import model.component.Port;
+import model.component.Endpoint;
+
+/**
+ * ...
+ * @author AdvaitTrivedi
+ */
+class DeleteSelectionCommand extends AbstractCommand 
+{
+	var selectionModel: SelectionModel;
+	var selectedComponents = new Array<Component>();
+	var selectedLinks = new Array<Link>();
+	var selectedPorts = new Array<Port>();
+	var selectedEndpoints = new Array<Endpoint>();
+
+	public function new(circuitDiagram: CircuitDiagramI, selectionModel: SelectionModel) 
+	{
+		this.setCircuitDiagram(circuitDiagram);
+		this.selectionModel = selectionModel;
+		this.selectedComponents = selectionModel.getComponents();
```

Comment:
So how would I do that? One way is to change those routines to return copies of the internal arrays (the copy could be a set) instead of the original array.  Another would be to have them return iterators and then, in this class, convert the iterators to new arrays or sets.  I couldn't find a library routine that takes an iterator and returns an array, but such a thing is easy to make.